uk.co.compucorp.civicrm.tasksassignments/js/test/tasks-notification-badge/component/tasks-notification-badge.component.spec.js

```diff
@@ -0,0 +1,52 @@
+/* eslint-env amd, jasmine */
+
+define([
+  'common/angularMocks',
+  'tasks-notification-badge/app'
+], function () {
+  'use strict';
+
+  describe('tasksNotificationBadge', function () {
+    var controller, $componentController, $log, $rootScope;
+
+    beforeEach(module('tasks-notification-badge'));
+    beforeEach(inject(function (_$componentController_, _$log_, _$rootScope_) {
+      $componentController = _$componentController_;
+      $log = _$log_;
+      $rootScope = _$rootScope_;
+      spyOn($log, 'debug');
+      spyOn($rootScope, '$emit');
+
+      compileComponent();
+    }));
+
+    it('is initialized', function () {
+      expect($log.debug).toHaveBeenCalled();
+    });
+
+    it('filters the Tasks where status is not completed and assignee is the logged in user', function () {
+      expect(controller.filters[0]).toEqual({
+        apiName: 'Task',
+        params: {
+          status_id: { '!=': 'Completed' },
+          assignee_contact_id: window.Drupal.settings.currentCiviCRMUserId
```

Comment:
Although technically correct, you better include the Session service and mock the response, since this is what you are ultimately using inside the badge implementation. If the session service happens to change and uses another method to determine the session it would break this test.